### PR TITLE
fix(backend/pandas): Convert target column to datetime before extracting datetime information TCTC-4825

### PR DESCRIPTION
**Pull request by lukapeschke**
closes TCTC-4823

Signed-off-by: Luka Peschke <luka.peschke@toucantoco.com>

### CODE REVIEW

File: server/src/weaverbird/backends/pandas_executor/steps/date_extract.py [hunk 1]
```diff
@@ -102,45 +107,45 @@ def execute_date_extract(
...
         elif dt_info == "firstDayOfPreviousQuarter":
-            first_month_of_quarter = 3 * ((serie_dt.month - 1) // 3) + 1
+            first_month_of_quarter = 3 * ((series_dt_accessor.month - 1) // 3) + 1
             first_month_of_prev_q = first_month_of_quarter - 3
             first_month_of_prev_q = first_month_of_prev_q.replace({-2: 10})
             result = to_datetime(
                 DataFrame(
                     {
-                        "year": serie_dt.year - (first_month_of_prev_q == 10),
+                        "year": series_dt_accessor.year - (first_month_of_prev_q == 10),
```
Comment (PrettyWood) [inline server/src/weaverbird/backends/pandas_executor/steps/date_extract.py:116]: I don't like the usage of `True == 1 / False == 0` here. It's not golf programming.
Not related to this issue but we should probably rewrite this piece of code to be explicit on the why we have 10, and when we want to do what

File: server/tests/steps/test_date_extract.py [hunk 2]
```diff
@@ -260,10 +248,7 @@ def test_date_extract_no_uint32(sample_df: DataFrame):
 
 
 def test_benchmark_dateextract(benchmark):
-    dates = [
-        datetime.datetime.today() + datetime.timedelta(days=nb_day)
-        for nb_day in list(range(1, 2001))
-    ]
+    dates = [datetime.today() + timedelta(days=nb_day) for nb_day in list(range(1, 2001))]
```
Comment (PrettyWood) [inline server/tests/steps/test_date_extract.py:251]:
```suggestion
    dates = [datetime.today() + timedelta(days=nb_day) for nb_day in range(1, 2001)]
```

**Review by PrettyWood — APPROVED** [review]
LGTM! I approve because my comments are not related to the fix but one of them can be included in this patch

File: server/src/weaverbird/backends/pandas_executor/steps/date_extract.py (hunk 1 above)
Comment (lukapeschke) [reply server/src/weaverbird/backends/pandas_executor/steps/date_extract.py:116]: Agreed, I'll submit a master-only patch for that

Comment (lukapeschke) [reply server/src/weaverbird/backends/pandas_executor/steps/date_extract.py:116]: https://github.com/ToucanToco/weaverbird/pull/1669

**Pull request merged**